ucp/_libs/utils.pyx

```diff
@@ -12,12 +12,70 @@ from cpython.memoryview cimport (
 from cython cimport boundscheck, wraparound
 from libc.stdint cimport uintptr_t
 
+try:
+    from numpy import dtype as numpy_dtype
+except ImportError:
+    numpy_dtype = None
+
 
 cdef struct iface_data_t:
     uintptr_t ptr
     bint readonly
 
 
+cdef dict itemsize_mapping = {
+    "|b1": 1,
+    "|i1": 1,
+    "|u1": 1,
+    "<i2": 2,
+    ">i2": 2,
+    "<u2": 2,
+    ">u2": 2,
+    "<i4": 4,
+    ">i4": 4,
+    "<u4": 4,
+    ">u4": 4,
+    "<i8": 8,
+    ">i8": 8,
+    "<u8": 8,
+    ">u8": 8,
+    "<f2": 2,
+    ">f2": 2,
+    "<f4": 4,
+    ">f4": 4,
+    "<f8": 8,
+    ">f8": 8,
+    "<f16": 16,
+    ">f16": 16,
+    "<c8": 8,
+    ">c8": 8,
+    "<c16": 16,
+    ">c16": 16,
+    "<c32": 32,
+    ">c32": 32,
+}
+
+cpdef Py_ssize_t get_itemsize(str fmt) except *:
+    """
+    Get the itemsize of the format provided.
+    """
+    if fmt is None:
+        raise ValueError("Expected `str`, but got `None`")
+    elif fmt == "":
+        raise ValueError("Got unexpected empty `str`")
```

Comment:
Why do we need to special case the empty string, wouldn't that be covered by the `ValueError` exception below?

Reply:
I'm not sure that error message would fit. An empty string would still raise even if NumPy was installed. So it's not some more complicated format (like `<M8[D]`) that we don't include in the `dict` above.

Reply:
Fair enough, thanks for clarifying.